llvm/lib/CodeGen/CalcSpillWeights.cpp

```diff
@@ -124,6 +127,75 @@ bool VirtRegAuxInfo::isRematerializable(const LiveInterval &LI,
 
     if (!TII.isTriviallyReMaterializable(*MI))
       return false;
+
+    VNIDefs[OrigVNI->id] = MI;
+  }
+
+  // If MI has register uses, it will only be rematerializable if its uses are
+  // also live at the indices it will be rematerialized at.
+  for (MachineOperand &Use : MRI.use_nodbg_operands(LI.reg())) {
+    if (Use.isUndef())
+      continue;
```

Comment:
This will miss uses in defs, this should use reg_nodbg_operands and check MO.readsReg

Reply:
Thanks, TIL.

> A sub-register def implicitly reads the other parts of the register being redefined unless the <undef> flag is set.